src/main/java/edu/byu/cs/autograder/compile/CompileHelper.java

```diff
@@ -66,7 +67,18 @@ private void packageRepo() throws GradingException {
         try {
             ProcessUtils.ProcessOutput output = ProcessUtils.runProcess(processBuilder, 90000); //90 seconds
             if (output.statusCode() != 0) {
-                Rubric.Results results = Rubric.Results.textError("Your Java source code could not be compiled", getMavenError(output.stdOut()));
+                String mavenError = getMavenError(output.stdOut());
+                String notes = "Your Java source code could not be compiled. See Details.";
+
+                Set<String> extraImports = NoExtraImportsVerifier.findExtraImports(mavenError);
+                if (!extraImports.isEmpty()) {
+                    notes += " It seems that you have imports from packages you cannot use. " +
+                            "Double check the Phase specs for the proper packages to import. " +
+                            "Please remove the following packages/imports from your code: " +
+                            String.join(", ", extraImports) + "." ;
```

Comment:
This does not contain a mention of where specifically they imported something improperly. A student who mistakenly imported server classes in their client could possibly read this and think they cannot use anything from their server anywhere (including elsewhere in the server, or their unit tests, both of which are allowed). Do you think this could output where the bad import is from, or perhaps ignore bad imports from packages that do exist in their code?

Reply:
Good catch. I forgot about your module independence verifier and currently there is duplication. 

> This does not contain a mention of where specifically they imported something improperly.

The mention of where something was imported improperly is given in the actual Maven output in the text results and I omitted that as I did not want to put too much text inside that little notes box and it would have more or less be copy and pasted from the actual Maven error which students can see if they just click the details button.

> A student who mistakenly imported server classes in their client could possibly read this and think they cannot use anything from their server anywhere (including elsewhere in the server, or their unit tests, both of which are allowed).

I can reword the message and rework the check to emphasize that this is meant to be talking about external libraries and not the modules inside the project.

> Do you think this could output where the bad import is from, or _perhaps ignore bad imports from packages that do exist in their code?_

I'm not sure what you mean by bad imports and ignoring them. Can you clarify?

Reply:
Oh, I think by bad imports you mean cases where the client imports packages from the server or vice versa.